pkg/background/update_request_controller.go

```diff
@@ -181,62 +162,55 @@ func (c *Controller) syncUpdateRequest(key string) error {
 	}
 	ur, err := c.urLister.Get(urName)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
```

Comment:
Quick question - we'll never get `IsNotFound` error if ur does not exist in the cache? I recall you had mentioned this in a previous PR, couldn't recall the exact case.

Reply:
I see. Returning `nil` also drops UR from the queue in `handleErr`, but without logging message.